src/controllers/productController.js

```diff
@@ -91,5 +91,78 @@ class ProductController {
       });
     }
   }
+
+  // A SELLER SEE ALL HIS PRODUCTS
+  static async myCollectionProducts(req, res) {
+    try {
+      const { id } = res.locals;
+      const products = await product.findAll({ where: { userId: id } });
+      if (products.length <= 0) {
+        return res.status(404).json({
+          ok: false,
+          message: 'You have no product in your collection',
+        });
+      }
+      return res.status(200).json({
+        ok: true,
+        message: ` ${products.length} products found`,
+        data: products,
+      });
+    } catch (error) {
+      return res.status(500).json({
+        status: 'Getting product failure',
+        message: error.message,
+      });
+    }
+  }
+
+  // USER CAN GET ALL AVAILABLE PRODUCTS
+  static async allAvailableProducts(req, res) {
+    try {
+      const availableProducts = await product.findAll({
+        where: { isAvailable: true },
+      });
+      if (availableProducts < 1) {
+        return res.status(409).json({
```

Comment:
I believe a more appropriate response status code would be `200` since the user request was processed successfully and the server returned a response. The response would be an empty array, but it is still a success response.

Reply:
Okaay! let me change it then!